zipline/pipeline/loaders/quarter_estimates.py

```diff
@@ -0,0 +1,582 @@
+            Number of quarters to go out in the future.
+        stacked_last_per_qtr : pd.DataFrame
+            A DataFrame with index of calendar dates, sid, and normalized
+            quarters with each row being the latest estimate for the row's
+            index values, sorted by event date.
+
+        Returns
+        -------
+        next_releases_per_date_index : pd.MultiIndex
+            An index of calendar dates, sid, and normalized quarters, for only
+            the rows that have a next event.
+        """
+
+        # We reset the index here because in pandas3, a groupby on the index
+        # will set the index to just the items in the groupby, so we will lose
+        # the normalized quarters.
+        next_releases_per_date = stacked_last_per_qtr.loc[
+            stacked_last_per_qtr[EVENT_DATE_FIELD_NAME] >=
+            stacked_last_per_qtr.index.get_level_values(SIMULTATION_DATES)
+        ].reset_index(NORMALIZED_QUARTERS).groupby(
+            level=[SIMULTATION_DATES, SID_FIELD_NAME]
+        ).nth(0).set_index(NORMALIZED_QUARTERS, append=True)
+        return next_releases_per_date.index
+
+
+class PreviousQuartersEstimatesLoader(QuarterEstimatesLoader):
+    def create_overwrite_for_estimate(self,
+                                      column,
+                                      column_name,
+                                      dates,
+                                      next_qtr_start_idx,
+                                      overwrite,
+                                      requested_quarter,
+                                      sid,
+                                      sid_idx):
+        return self.overwrite_with_null(column,
+                                        dates,
+                                        next_qtr_start_idx,
+                                        overwrite,
+                                        sid_idx)
+
+    def get_shifted_qtrs(self, zero_qtrs, num_quarters):
+        return zero_qtrs - (num_quarters - 1)
+
+    def get_zeroth_quarter_idx(self, num_quarters, stacked_last_per_qtr):
+        """
+        Filters for releases that are on or after each simulation date and
+        determines the previous quarter by picking out the most recent
+        release relative to each date in the index.
+
+        Parameters
+        ----------
+        num_quarters : int
+            Number of quarters to go out in the past.
+        stacked_last_per_qtr : pd.DataFrame
+            A DataFrame with index of calendar dates, sid, and normalized
+            quarters with each row being the latest estimate for the row's
+            index values, sorted by event date.
+
+        Returns
+        -------
+        previous_releases_per_date_index : pd.MultiIndex
+            An index of calendar dates, sid, and normalized quarters, for only
+            the rows that have a previous event.
+        """
+
+        # We reset the index here because in pandas3, a groupby on the index
```

Comment:
I probably meant some particular version of pandas, but unsure of what it is anymore. At any rate, `as_index=False` works.